pkg/skaffold/runner/runner.go

```diff
@@ -260,6 +266,13 @@ func (r *SkaffoldRunner) BuildAndTest(ctx context.Context, out io.Writer, artifa
 	return bRes, err
 }
 
+// Deploy deployes the given artifacts
+func (r *SkaffoldRunner) Deploy(ctx context.Context, out io.Writer, artifacts []build.Artifact) ([]deploy.Artifact, error) {
+	dRes, err := r.Deployer.Deploy(ctx, out, artifacts)
+	r.hasDeployed = true
```

Comment:
I guess it's not different than what we had before, but if `Deploy()` errors out do we want to set this flag?

Reply:
I guess it could fail and have deployed a few things. We want to clean this up

Reply:
yeah, not really a good way to know whether or not that's the case so let's just be safe.